### PR TITLE
Add hooks

### PR DESCRIPTION
**Pull request by Laylo309**
- [x] Refactor the Calculator component (components/Calculator.js) from a class-based component to a functional component with hooks.

- [x] After the refactor is done, test the app in the browser. It should keep the same functionality.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,36 +1,27 @@
...
-    this.handleEvent = this.handleEvent.bind(this);
-  }
+const Calculator = () => {
+  const [calculator, addState] = useState({ total: 0, next: null, operation: null });
+  const { total, next, operation } = calculator;
 
-  handleEvent = (e) => {
+  const handleEvent = (e) => {
     e.preventDefault();
-    this.setState((prev) => calculate(prev, e.target.name));
```
Comment (george-shammar) [inline src/components/Calculator.js:19]: - [ ]  Your application works and looks nice already. However, when I double-clicked on an operator, I got this browser error: `TypeError: one.times is not a function`. Please, let's include an exception to handle this error.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @Laylo309 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (Laylo309) [reply src/components/Calculator.js:19]: Done ✅

**Review by Georjane — APPROVED** [review]
## Status: PR Approved 💯 🥇 
#### Hello @Laylo309  

Your project is complete!  You have met all requirements for this project. It is a thumbs up for me. 👍🏽 You can go ahead and merge this PR. 
Congratulations! 🎉 👏👏

_Good luck with your next project and Happy coding!_ 💻

**Pull request merged**